shared/lib_irradproc.h

```diff
@@ -628,9 +628,68 @@ double sun_rise_and_set(double* m_rts, double* h_rts, double* delta_prime, doubl
 * \param[out] needed_values[7] zenith topocentric zenith angle (degrees)
 * \param[out] needed_values[8] azimuth topocentric azimuth angle (degrees)
 */
+
+
 void calculate_spa(double jd, double lat, double lng, double alt, double pressure, double temp,
     double delta_t, double tilt, double azm_rotation, double ascension_and_declination[2], double needed_values[9]);
 
+/**
+ * Table for storing the recently computed solarpos_spa intermediate outputs
+ * The algorithm reuses the outputs from the last 3 days or so, so the hash table is emptied every 3 days to reduce size
+ * Latitude, Longitude, Altitude, Tilt and Azimuth are not in the key because they remain constant throughout
+ */
+struct spa_table_key {
+    double jd;
+    double delta_t;
+    int pressure;
+    int temp;
+    // these are both inputs and outputs (e.g. also stored in the output vector)
+    double ascension;
+    double declination;
+
+    bool operator==(const spa_table_key &other) const
+        { return (jd == other.jd
+                && delta_t == other.delta_t
+                && pressure == other.pressure
+                && temp == other.temp
+                && ascension == other.ascension
+                && declination == other.declination
+                );
+        }
+
+    spa_table_key(double j, double dt, double p, double t, double a, double d):
+        jd(j), delta_t(dt), ascension(a), declination(d)
+    {
+        int pressure_bucket = 10;
```

Comment:
Do you have a sensitivity plot based on pressure and temp bucket size?

Reply:
Sure, the sensitivity is super low to pressure and temperature:

pressure bucket | temp bucket | time (ms) | result change
-- | -- | -- | --
10 | 5 | 3250 | na
2 | 5 | 3443 | na
2 | 2 | 3490 | na
10 | 2 | 3411 | na
1 | 1 | 3630 | na
20 | 10 | 3238 | na